src/js/controller/controller.js

```diff
@@ -924,6 +934,25 @@ Object.assign(Controller.prototype, {
             const wrappedIndex = wrapPlaylistIndex(index, length);
 
             return _programController.setActiveItem(wrappedIndex).catch(error => {
+                if (error.code === ASYNC_PLAYLIST_ITEM_REJECTED) {
+                    // If all items were rejected throw. This will fail setup with setupError code 102700
+                    const allSkipped = _programController.asyncItems.reduce((skipped, asyncItem) =>
+                        skipped && asyncItem.skipped, true);
+                    if (allSkipped) {
+                        throw error;
+                    }
+                    // If the last item is rejected we need to stop and complete playlist playback. Otherwise,
```

Comment:
is this true in a recommendations case? if the last item is rejected and we have recs, shouldn't it move onto those or no?

Reply:
We can't because "nextUp" is not updated without a playlist item event. This would require changes to related and nextUp which I am not doing as part of this 2 pointer. (Says so below and in the ticket).

Reply:
Yeah if "nextUp" was the recommendation we'd be fine, but it's not because the "playlistItem" event for the last item is what updates "nextUp" and it's been skipped. The logic that updates "nextUp" needs to be handled either here in the controller and/or in the program-controller to work in sync with this system. Or, we need a new event to signal an item was skipped, to update related.